[Wrath] WI-VARIANCE-BUILD-FIX: pre-existing TS build errors in @yakcc/variance (gates pr-ci.yml)

**Sister:** open — Wrath or sister, depending on availability
**Track:** Followup to WI-CI-FAST-PATH Phase 1 (#196)
**Depends on:** WI-LINT-DRIFT-FIX (#235, PR #236) — landed; surfaced this issue

## The drift

After PR #236 landed (lint drift fix), pr-ci.yml run 25622878487 on that PR's own merge commit FAILED with TypeScript build errors in `@yakcc/variance`. These are pre-existing on main; the new fast CI just surfaces them for the first time.

### Errors observed in CI

`packages/variance/src/index.ts`:
- TS2307: Cannot find module '@yakcc/contracts' (lines 23, 39 of index.ts; line 23 of index.props.ts) — workspace dependency resolution issue specific to CI runner
- TS2345 (multiple): `Set<unknown>` not assignable to `ReadonlySet<string>` — strict-mode iteration without explicit type
- TS7006 (multiple): Parameter 'e' implicitly has 'any' type — missing parameter type annotations

### Errors observed locally

`pnpm --filter @yakcc/variance build`:
- TS2688: Cannot find type definition file for 'node' — `@types/node` missing from variance's dependencies

So locally and in CI, the package has ≥2 different failure modes:
1. Missing `@types/node` (local)
2. Workspace dep resolution + strict-mode type errors (CI-only manifestation)

## Why this matters

CI is correctly gating PRs on this. Until it's fixed:
- Every Wrath/sister PR fails pr-ci.yml on `build + test (affected packages)` if any package they touch transitively depends on @yakcc/variance
- The full benefit of #196 Phase 1 ("≤5min PR feedback") is undermined because feedback says "failed" rather than "passed"

## Investigation hints

- `packages/variance/package.json` — verify dev dependencies (`@types/node` likely needed)
- `packages/variance/tsconfig.json` — verify it extends from a base with strict settings; verify references to @yakcc/contracts
- `packages/variance/src/index.ts` — fix Set<unknown>→Set<string> iterations and add explicit type annotations to forEach callbacks (lines 272, 299, 300, 301, 304, 307, 309, 371)
- `packages/variance/src/index.props.ts` line 23 — same import resolution issue
- Check whether sister @yakcc/seeds and others have similar pre-existing type errors (typecheck full-workspace will surface them all)

## Acceptance

- `pnpm --filter @yakcc/variance build` clean locally
- `pnpm -r typecheck` clean across all packages
- Open a fresh PR and verify pr-ci.yml passes its build+test job

## Out of scope

- Any source semantic change beyond minimum needed to satisfy strict TS
- Biome lint changes (already covered by #235)
- Other packages with similar drift — file separate issues per package

cc: parent #196

--- a/packages/variance/package.json
+++ b/packages/variance/package.json
@@ -19,7 +19,7 @@
     "test:coverage": "vitest run --coverage",
     "lint": "biome check src/"
   },
-  "peerDependencies": {
+  "dependencies": {
     "@yakcc/contracts": "workspace:*"
   },
   "devDependencies": {

--- a/packages/variance/src/index.ts
+++ b/packages/variance/src/index.ts
@@ -267,8 +267,8 @@ function scoreSecurity(canonical: SpecYak, candidate: SpecYak): number {
  * Both-empty → 1.0 (both declare no output guarantees; fully aligned).
  */
 function scoreBehavioral(canonical: SpecYak, candidate: SpecYak): number {
-  const a = new Set(canonical.postconditions.map(normalize));
-  const b = new Set(candidate.postconditions.map(normalize));
+  const a = new Set<string>(canonical.postconditions.map(normalize));
+  const b = new Set<string>(candidate.postconditions.map(normalize));
   return jaccard(a, b);
 }
 
@@ -296,15 +296,20 @@ function scoreErrorHandling(canonical: SpecYak, candidate: SpecYak): number {
   if (cErrs === undefined || cErrs.length === 0) return 0.0;
   if (dErrs === undefined || dErrs.length === 0) return 0.0;
 
-  const descA = new Set(cErrs.map((e) => normalize(e.description)));
-  const descB = new Set(dErrs.map((e) => normalize(e.description)));
+  type ErrorCond = { readonly description: string; readonly errorType?: string | undefined };
+  const descA = new Set<string>(cErrs.map((e: ErrorCond) => normalize(e.description)));
+  const descB = new Set<string>(dErrs.map((e: ErrorCond) => normalize(e.description)));
   const descScore = jaccard(descA, descB);
 
-  const typeA = new Set(
-    cErrs.filter((e) => e.errorType !== undefined).map((e) => normalize(e.errorType as string)),
+  const typeA = new Set<string>(
+    cErrs
+      .filter((e: ErrorCond) => e.errorType !== undefined)
+      .map((e: ErrorCond) => normalize(e.errorType as string)),
   );
-  const typeB = new Set(
-    dErrs.filter((e) => e.errorType !== undefined).map((e) => normalize(e.errorType as string)),
+  const typeB = new Set<string>(
+    dErrs
+      .filter((e: ErrorCond) => e.errorType !== undefined)
+      .map((e: ErrorCond) => normalize(e.errorType as string)),
   );
   const typeScore = jaccard(typeA, typeB);
 
@@ -363,10 +368,10 @@ function scoreInterface(canonical: SpecYak, candidate: SpecYak): number {
   const paramKey = (p: { readonly name: string; readonly type: string }) =>
     `${normalize(p.name)}::${normalize(p.type)}`;
 
-  const inA = new Set(canonical.inputs.map(paramKey));
-  const inB = new Set(candidate.inputs.map(paramKey));
-  const outA = new Set(canonical.outputs.map(paramKey));
-  const outB = new Set(candidate.outputs.map(paramKey));
+  const inA = new Set<string>(canonical.inputs.map(paramKey));
+  const inB = new Set<string>(candidate.inputs.map(paramKey));
+  const outA = new Set<string>(canonical.outputs.map(paramKey));
+  const outB = new Set<string>(candidate.outputs.map(paramKey));
 
   return 0.5 * jaccard(inA, inB) + 0.5 * jaccard(outA, outB);
 }

